File: Exercism Codes/PHP/ListOps.php
<?php

declare(strict_types=1);

class ListOps
{
    public function append(array $list1, array $list2): array
    {
        return array_merge($list1, $list2);
    }

    public function concat(array $list1, array ...$listn): array
    {
        return array_merge($list1, ...$listn);
    }

    public function filter(callable $predicate, array $list): array
    {
        return array_values(array_filter($list, $predicate));
    }

    public function length(array $list): int
    {
        return count($list);
    }

    public function map(callable $function, array $list): array
    {
        return array_map($function, $list);
    }

    public function foldl(callable $function, array $list, $accumulator)
    {
        for ($i = 0; $i < count($list); $i++) {
            $accumulator = $function($accumulator, $list[$i]);
        }
        return $accumulator;
    }

    public function foldr(callable $function, array $list, $accumulator)
    {
        for ($i = count($list) - 1; $i >= 0; $i--) {
            $accumulator = $function($accumulator, $list[$i]);
        }
        return $accumulator;
    }

    public function reverse(array $list): array
    {
        return array_reverse($list);
    }
}

File: Exercism Codes/C#/ListOps.cs
public static class ListOps
{
    public static int Length<T>(List<T> input)
    {
        return input.Count;
    }

    public static List<T> Reverse<T>(List<T> input)
    {
        return input.Reverse<T>().ToList();
    }

    public static List<TOut> Map<TIn, TOut>(List<TIn> input, Func<TIn, TOut> map)
    {
        return input.Select(map).ToList();
    }

    public static List<T> Filter<T>(List<T> input, Func<T, bool> predicate)
    {
        return input.Where(predicate).ToList();
    }

    public static TOut Foldl<TIn, TOut>(List<TIn> input, TOut start, Func<TOut, TIn, TOut> func)
    {
        return input.Aggregate(start, func);
    }

    public static TOut Foldr<TIn, TOut>(List<TIn> input, TOut start, Func<TIn, TOut, TOut> func)
    {
        for (int i = input.Count - 1; i >= 0; i--)
        {
            start = func(input[i], start);
        }
        return start;
    }

    public static List<T> Concat<T>(List<List<T>> input)
    {
        return input.SelectMany(x => x).ToList();
    }

    public static List<T> Append<T>(List<T> left, List<T> right)
    {
        return left.Concat(right).ToList();
    }
}